src/wallet/hdwallet.ts

```diff
@@ -225,16 +224,24 @@ export abstract class HDWallet extends BaseWallet implements HDNodeLike<HDWallet
             operation: 'extendedKey',
         });
 
-        return encodeBase58Check(
+        const myKey = encodeBase58Check(
             concat([
                 '0x0488ADE4',
-                zpad(this.depth, 1),
+                '0x' + zpad(this.depth, 2),
                 this.accountFingerprint ?? '',
-                zpad(this.index, 4),
+                '0x' + zpad(this.index, 4),
                 this.chainCode,
                 concat(['0x00', this.privateKey]),
             ]),
         );
+        return myKey;
+    }
```

Comment:
Since we are planning to revert this HDWallet class back to the original ethers HDNodeWallet class lets plan to move to these fixes over after that. I will keep this PR open until then.